allen-cell-animated/nucmorph-colorizer
# feat: Physical time

Problem
=======
Closes #61, physical time!

Solution
========
- Adds new dataset metadata parameters (start time + frame duration). The default nucmorph datasets have been updated to use these values.
- Plumbing code to pass parameters + config down to the `CanvasOverlay`.
- Rendering + formatting for the new timestamp.
- Changes the rendering pattern for `CanvasOverlay`.
  - A lot of these changes needed to happen to allow the overlay elements (scale bar + timestamp) to have a transparent background beneath them. The biggest change is delaying element rendering, so that the size of the elements can be summed and a backdrop can be drawn first beneath them.

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open preview link (https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-149/)
2. Make sure the "Show timestamp" checkbox is enabled.
3. Play, and observe the timestamp updating.
4. Also, try disabling/enabling the scale bar and timestamp. The background will disappear.

Screenshots (optional):
-----------------------
![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/d6ef2e60-3605-4ccd-88c7-8ce39fd5c684)

### Video Talk-through (w/ Audio)

https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/200feb1a-0554-4fe2-986d-7bfe4fcc222c

Addendum video about different timestamp formats:

https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/28943b34-2e1f-45fd-9ae9-2c3d364a23da

Thanks for contributing!
